pkg/apis/app/v2/kieapp_types.go

```diff
@@ -152,18 +152,32 @@ type KieServerSet struct {
 	KafkaJbpmEventEmitters *KafkaJBPMEventEmittersObject `json:"kafkaJbpmEventEmitters,omitempty"`
 	Cors                   *CORSFiltersObject            `json:"cors,omitempty"`
 	// MDBMaxSession number of KIE Executor sessions
-	MDBMaxSession *int `json:"MDBMaxSession,omitempty"`
+	MDBMaxSession    *int              `json:"MDBMaxSession,omitempty"`
+	TerminationRoute TerminationObject `json:"termination,omitempty"`
+}
+
+type TerminationObject struct {
+	// If set to true disableSSL will be skipped for the console the target port will be http and the termination will be edge
+	EnableInsecureEdgeTermination bool `json:"enableInsecureEdgeTermination,omitempty"`
+	// +kubebuilder:validation:Enum:=http;https
+	// Protocol to use, if no value is provided, https is the default protocol, if the EnableInsecureEdgeTermination is true http will be used
+	Protocol string `json:"protocol,omitempty"`
+	// +kubebuilder:validation:Enum:=passthrough;edge
+	// Smart Router termination, if disableSSL is true and no value is provided, passthrough is the default protocol, otherwise edge could be used.
+	Termination      string `json:"termination,omitempty"`
+	EnableTlsSection bool   `json:"enableTlS,omitempty"`
 }
 
 // ConsoleObject configuration of the RHPAM workbench
 type ConsoleObject struct {
-	KieAppObject `json:",inline"`
-	SSOClient    *SSOAuthClient     `json:"ssoClient,omitempty"`
-	GitHooks     *GitHooksVolume    `json:"gitHooks,omitempty"`
-	Jvm          *JvmObject         `json:"jvm,omitempty"`
-	PvSize       string             `json:"pvSize,omitempty"`
-	Cors         *CORSFiltersObject `json:"cors,omitempty"`
-	DataGridAuth *DataGridAuth      `json:"dataGridAuth,omitempty"`
+	KieAppObject     `json:",inline"`
+	SSOClient        *SSOAuthClient     `json:"ssoClient,omitempty"`
+	GitHooks         *GitHooksVolume    `json:"gitHooks,omitempty"`
+	Jvm              *JvmObject         `json:"jvm,omitempty"`
+	PvSize           string             `json:"pvSize,omitempty"`
+	Cors             *CORSFiltersObject `json:"cors,omitempty"`
+	DataGridAuth     *DataGridAuth      `json:"dataGridAuth,omitempty"`
+	TerminationRoute TerminationObject  `json:"termination,omitempty"`
```

Comment:
Can we somehow improve naming for KieApp? We have in ConsoleObject TerminationRoute that is represent in KieApp as termination. And then we also have in TerminationObject string variable Termination that is represent in KieApp as property called termination. This may look confusing and users may do some easy mistakes in KieApp.
E.g. wrongly set it up edge termination
"termination: edge
..."
instead of
"termination:
     termination: edge
     ..."

Reply:
Improved with TerminationRoute on TerminationObject and on json